### ISSUE TITLE
Lag for window button operations (close, maximize, minimize)

### ISSUE THREAD

**Issue by mathaou** [bitsdojo/bitsdojo_window#187]
Flutter 3.1.0-0.0.pre.1887 (upgraded from 3.0.1 to see if latest master could help)

With a setup like this

```dart
final _windowButtons = [
  MinimizeWindowButton(
    animate: false,
    colors: WindowButtonColors(iconNormal: Colors.black),
  ),
  MaximizeWindowButton(
    animate: false,
    colors: WindowButtonColors(iconNormal: Colors.black),
  ),
  CloseWindowButton(
    animate: false,
    colors: WindowButtonColors(
        iconNormal: Colors.black,
        iconMouseDown: Colors.white,
        iconMouseOver: Colors.white,
        mouseDown: Colors.red,
        mouseOver: Colors.red),
  )
];
```

Using any of these operations (or calling the same functions from the window_manager plugin) has a pretty jarring delay before the operation completes. On the run-of-the-mill native app all of those are practically instantaneous, but profiling the code shows me that I'm spending close to 70ms getting a GestureDetector to service an action, and then pass that through some kind of ffi that needs bytestreams or some other binary blob to function. My gut feeling is that there isn't much that can be done to speed this up, but I thought I'd make an issue in case this is thoroughly a "me" problem and I'm just doing something wrong.

**Comment by ngugcx** [comment]
I experienced a similar problem today, not sure if it's related.
I put WindowButtons under MoveWindow which enables onDoubleTap event in a Stack.
In Flutter, if onDoubleTap is enabled, onTap is delayed for some time (300ms?) to detect double tap. [https://github.com/flutter/flutter/issues/110300](url)
```
Stack(children: [
  Row(
    children: [
      Expanded(
          child: Tabbar()),
      const WindowButtons()
    ],
  ),
  MoveWindow(),
])
```

**Comment by Auties00** [comment]
> MoveWind

Noticed the same exact thing

**Comment by bitsdojo** [comment]
Thank you all for reporting. Will investigate.

**Comment by bitsdojo** [comment]
What desktop OS are you using for testing?

**Comment by mathaou** [comment]
> What OS are you using for testing?

I used Windows.

**Comment by nickcuypers** [comment]
yeah I experienced the same today, also on Windows OS btw double tapping the app bar is "instant" and tapping the buttons is with a slight delay, I will try and take the buttons out of the moveWindow widget for now

**Comment by Auties00** [comment]
> yeah I experienced the same today, also on Windows OS btw double tapping the app bar is "instant" and tapping the buttons is with a slight delay, I will try and take the buttons out of the moveWindow widget for now

Just use the stack as previously explained by @ngugcx

**Comment by nickcuypers** [comment]
> > yeah I experienced the same today, also on Windows OS btw double tapping the app bar is "instant" and tapping the buttons is with a slight delay, I will try and take the buttons out of the moveWindow widget for now
> 
> Just use the stack as previously explained by @ngugcx

yes was exactly what I did 👍

**Comment by nickcuypers** [comment]
is there any news or estimation on this issue? workaround aint perfect for my needs sadly

**Comment by Deava0** [comment]
I have the same issue, developed multiple apps and experienced it in all of them,
any chance we're getting a fix anytime soon ?

`
actions: MoveWindow(
            child: Row(
              children: [
                const Spacer(),
                MinimizeWindowButton(
                  colors: buttonColors,
                ),
                MaximizeWindowButton(
                  colors: buttonColors,
                ),
                CloseWindowButton(
                  colors: buttonColorsExit,
                )
              ],
            ),
          ),
`

**Comment by zaher-neon** [comment]
the issue is something related to MoveWindow() widget.
when i added the MoveWindow() on top of the window buttons using Stack() it made them slow.
